### ISSUE TITLE
proposal: missing_alt_text

### ISSUE THREAD

**Issue by gregorscholz** [dart-lang/sdk#59178]
## Description

Image widget should be accessible for everybody and therefore contain an alt text.

## Details

For vision impaired people images can be annoying, they often use screen readers to get the information on the page. To also get information about the images, Image widgets should contain a semanticLabel.

## Bad Examples
```
Image(
    image: AssetImage('user_profile_picture'),
),
```

## Good Examples 

```
Image(
    image: AssetImage('user_profile_picture'),
    semanticLabel: 'profile picture',
),
```

## Discussion

With a semanticLabel added, people using screen readers can get a better overview of what information is displayed on the screen.
We as a community want to be open for everybody as much as we want be build applications which are accessible for everyone.

Also mentioned here: [2300](https://github.com/dart-lang/sdk/issues/58251)

**Comment by srawlins** [comment]
We won't be implementing this as a warning or a lint rule in the Dart analyzer. We now have a high bar for implementing and maintaining built-in static analysis rules, and this has too many false positives. As of Dart 3.10 (Flutter 3.38), a [new analyzer plugins system](https://dart.dev/tools/analyzer-plugins) is available, and the requested static analysis here could be implemented as an analyzer plugin.
